### PR TITLE
Initial support for Vue 3.

### LINKED ISSUE #51: Vue 3 support
Hello,

Is there a way to get this working within Vue 3?

Vue 3 no longer exports vue e.g. `import Vue from 'vue';` doesn't exist.

Is there a work around?

Thanks

### PR DESCRIPTION
**Pull request by joeldenning**
This pull request is backwards compatible and can be published as 1.9.0. Resolves #51

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,3 +1,12 @@
 {
-  "presets": ["@babel/preset-env"]
+  "presets": ["@babel/preset-env"],
+  "env": {
+    "test": {
+      "presets": [
+        ["@babel/preset-env", {
+          "targets": "current node"
```
Comment (joeldenning) [inline .babelrc:7]: This allows us to use async/await in tests without including regeneratorRuntime

File: src/single-spa-vue.js [hunk 2]
```diff
@@ -141,8 +151,12 @@ function update(opts, mountedInstances, props) {
 function unmount(opts, mountedInstances, props) {
   return Promise.resolve().then(() => {
     const instance = mountedInstances[props.name];
-    instance.vueInstance.$destroy();
-    instance.vueInstance.$el.innerHTML = "";
+    if (opts.createApp) {
+      instance.vueInstance.unmount(instance.domEl);
```
Comment (joeldenning) [inline src/single-spa-vue.js:155]: Vue 3 is better at unmounting! No need to clear out the innerHTML ourselves, and the router reportedly unmounts correctly in Vue 3

File: src/single-spa-vue.js [hunk 3]
```diff
@@ -112,11 +117,16 @@ function mount(opts, mountedInstances, props) {
       appOptions.data = {};
     }
 
-    appOptions.data = { ...appOptions.data, ...props };
+    appOptions.data = () => ({ ...appOptions.data, ...props });
```
Comment (joeldenning) [inline src/single-spa-vue.js:120]: Vue 3 requires the data option to be a function. Doing so also works in Vue 2

**Comment by joeldenning** [comment]
> One question, will vue-cli-plugin-single-spa also need to be updated to detect which version of Vue was used and modify the template accordingly?

Not from what I've seen. I created https://github.com/joeldenning/vue-3-single-spa by doing `vue create` and then `vue add single-spa` and the cli plugin worked without issue. I did create https://github.com/single-spa/vue-cli-plugin-single-spa/issues/15 to track a change we need to make so that the `main.js` file is automatically configured correctly based on the Vue version

**Comment by filoxo** [comment]
Ok cool. The reason I ask is because of `import * as Vue from 'vue'` ([this line from the demo repo](https://github.com/joeldenning/vue-3-single-spa/blob/master/src/main.js#L2) you put together) and I thought Vue had removed a default export so `import Vue from 'vue'` (which is generated in the [main.js file](https://github.com/single-spa/vue-cli-plugin-single-spa/blob/master/template/src/main.js#L2) from the cli plugin) wouldn't work. That'll been tested/solved in the plugin code anyway. I unfortunately don't have time to test these all right now but this code looks good to me!

**Comment by joeldenning** [comment]
> I thought Vue had removed a default export

Yes this is true. `vue add single-spa` doesn't throw an error, but it **does** create a `main.js` file that doesn't work with Vue 3. I had to modify the main.js to get that example working. Once we fix https://github.com/single-spa/vue-cli-plugin-single-spa/issues/15, it will all work though

**Pull request merged**